src/neuralnet/superblock.h

```diff
@@ -0,0 +1,248 @@
+#pragma once
+
+#include "scraper/fwd.h"
+#include <string>
+#include "key.h"
+#include "neuralnet/cpid.h"
+
+std::string UnpackBinarySuperblock(std::string block);
+std::string PackBinarySuperblock(std::string sBlock);
+
+namespace NN {
+
+// TODO: move this out into a common location...?
+template<typename integer_t>
+struct Compact
+{
+    Compact() : m_value(0)
+    {
+    }
+
+    Compact(integer_t value) : m_value(value)
+    {
+    }
+
+    integer_t m_value;
+
+    // TODO: implement the other general-purpose operators...
+
+    integer_t operator*() const
+    {
+        return m_value;
+    }
+
+    bool operator<(Compact<integer_t> other) const
+    {
+        return m_value < other.m_value;
+    }
+
+    bool operator<(integer_t other) const
+    {
+        return m_value < other;
+    }
+
+    void operator=(integer_t value)
+    {
+        m_value = value;
+    }
+
+    Compact<integer_t>& operator++()
+    {
+        m_value++;
+
+        return *this;
+    }
+
+    Compact<integer_t> operator++(int)
+    {
+        Compact<integer_t> original = *this;
+        ++*this;
+
+        return original;
+    }
+
+    unsigned int GetSerializeSize(int nType, int nVersion) const
+    {
+        if (m_value < 253) {
```

Comment:
Why don't we use `GetSizeOfCompactSize`, `WriteCompactSize` and `ReadCompactSize` from `serialize.h` for these?

Reply:
I wrote it like that at first, but those are specialized for deserializing container sizes and impose an upper bound of `MAX_SIZE` that's too low for us to use for some of the bigger stats values. For the experiment, I just copied the logic instead of trying to finagle a workaround for that.

Reply:
Seems like the only MAX_SIZE limit specialization is on ReadCompactSize vs Unserialize. The other two look identical.